ext/AvizExt/viz/scenarios.jl

```diff
@@ -64,24 +67,62 @@ function ADRIA.viz.scenarios!(
     # Set series colors
     merge!(series_opts, _get_series_opt_colors(rs, data, opts, series_opts))
 
+    if get(opts, :summarize, true)
+        _plot_scenarios_confint!(ax, rs, data)
+    else
+        _plot_scenarios_series!(ax, rs, data, series_opts)
+    end
+
+    # Plot Histograms when opts[:histogram] is true
+    get(opts, :histogram, true) && _plot_scenarios_hist(g, rs, data)
+
     # Render legend
-    _render_scenarios_legend(g, rs, opts)
+    legend_position = get(opts, :histogram, true) ? (1, 3) : (1, 2)
+    _render_scenarios_legend(g, rs, legend_position, opts)
 
-    _plot_scenarios_series!(ax, data, series_opts)
-    _plot_scenarios_hist(g, rs, data)
+    ax.xlabel = "Year"
+    # ax.ylabel = metric_label(metric)
 
     return g
 end
 
-function _plot_scenarios_series!(ax::Axis, data::NamedDimsArray, series_opts::Dict)::Nothing
-    series!(ax, data'; series_opts...)
+function _plot_scenarios_confint!(ax::Axis, rs::ResultSet, data::NamedDimsArray)::Nothing
+    x_timesteps::UnitRange{Int64} = 1:size(data, 1)
+    ordered_types = _order_by_variance(data, scenario_type(rs))
 
-    # ax.ylabel = metric_label(metric)
-    ax.xlabel = "Year"
+    selected_scenarios = [scenario_type(rs)[type] for type in ordered_types]
+    confints = [confint(data[:, scenario], :scenarios) for scenario in selected_scenarios]
+    colors = [scenario_colors(rs)[scenario][1][1] for scenario in selected_scenarios]
+
+    for idx in eachindex(ordered_types)
+        band_alpha = max(0.7 - idx * 0.1, 0.4)
+        band_color = (colors[idx], band_alpha)
+        y_lower, y_upper = confints[idx][1], confints[idx][3]
+        band!(ax, x_timesteps, y_lower, y_upper; color=band_color)
+    end
+
+    for idx in eachindex(ordered_types)
+        line_color = (colors[idx], 0.8)
+        y_median = confints[idx][2]
+        lines!(ax, y_median; color=line_color, linewidth=4)
+    end
 
     return nothing
 end
 
+function _plot_scenarios_series!(
+    ax::Axis, rs::ResultSet, data::NamedDimsArray, series_opts::Dict
+)::Nothing
+    series_colors = pop!(series_opts, :color)
+    for type in _order_by_variance(data, scenario_type(rs))
+        selected_scenarios = scenario_type(rs)[type]
```

Comment:
You're re-calling `scenario_type()` multiple times here. Best to do it once outside the loop.